### ISSUE TITLE
If evaluation data has only one class, how should we compute AUC?

### ISSUE THREAD

**Issue by yaeldekel** [dotnet/machinelearning#4713]
Currently if there is only data from one class, we throw an exception (see issue #4692 ).
It might make more sense to return NaN instead of that.

**Comment by justinormont** [comment]
I fully agree. This scenario keep arising. `NaN` is a much more friendly result than throwing an error. 

Even if AUC is returned as `NaN`, other metrics generally have values. Eg:
```
AUC:                NaN
Accuracy:           0.987654
Positive precision: 0.000000 (or NaN)
Positive recall:    0.000000 (or NaN)
Negative precision: 1.000000
Negative recall:    0.987654
F1 Score:           0.000000
```
When only negative labels are sent to the metric computation, and a few are incorrectly predicted as positive.

If none of the negative examples are predicted as positive (no false positives) are predicted, positive precision/recall would also be `NaN`, though we may be mapping these to `0.0`.
